registry/registry.go

```diff
@@ -61,6 +65,54 @@ func hasFile(files []os.FileInfo, name string) bool {
 	return false
 }
 
+// ReadCertsDirectory reads the directory for TLS certificates
+// including roots and certificate pairs and updates the
+// provided TLS configuration.
+func ReadCertsDirectory(tlsConfig *tls.Config, directory string) error {
```

Comment:
There is some mismatch with the logic to create TLS configurations for the docker cli to daemon and daemon to the registry. The config for the docker cli passes in the root and client certificates information, while the daemon attempts to interpret the names of the files in `/etc/docker/certs.d/<hostname>`.  This logic could certainly be moved to tlsconfig, but then there would be two different methods for creating TLS client configs inside that package. This change just pulls it out from inside a registry function and exposes it, we could move it completely out of registry as well. WDYT?